app/models/generic_object_definition.rb

```diff
@@ -7,6 +16,18 @@ class GenericObjectDefinition < ApplicationRecord
   has_many  :generic_objects
 
   def defined_attributes
-    @defined_attributes ||= properties[:attributes].stringify_keys
+    properties[:attributes]
+  end
+
+  def properties=(props)
+    props = props.symbolize_keys
+    if props.key?(:attributes)
+      # unify attribute name as string and attribute type as symbol
+      props[:attributes] = props[:attributes].each_with_object({}) do |(name, type), memo|
+        raise ArgumentError, "#{type} is not recognized type" unless TYPE_MAP.key?(type.to_sym)
+        memo[name.to_s] = type.to_sym
+      end
```

Comment:
I think this can be simplified(?) with:

```ruby
props[:attributes] = props[:attributes].stringify_keys.transform_values { |v| v.to_sym }
bad_types = props[:attributes].values.reject { |v| TYPE_MAP.key?(v) }
raise ArgumentError, "property types include some invalid valies: #{bad_types}" if bad_types.any?
```

Note that overridding